cartography/intel/crxcavator/crxcavator.py

```diff
@@ -0,0 +1,220 @@
+    # will build a cleaner object for import into graph
+
+    _data_index = 1
+    _row_index = 0
+
+    extensions = []
+    for extension in extension_json.items():
+        details = extension[_data_index][_row_index]
+        if not details:
+            continue
+        extension_id = details['extension_id']
+        version = details['version']
+        data = details['data']
+        # permissions warnings can be a list or null, if list concat to a single string
+        permission_warnings = data['webstore'].get('permission_warnings')
+        if permission_warnings:
+            permission_warnings = ', '.join(permission_warnings)
+        extensions.append({
+            'id': "{0}|{1}".format(extension_id, version),
+            'extension_id': extension_id,
+            'version': version,
+            'risk_total': data['risk'].get('total'),
+            'risk_metadata': json.dumps(data['risk'].get('metadata')),
+            'address': data['webstore'].get('address'),
+            'email': data['webstore'].get('email'),
+            'icon': data['webstore'].get('icon'),
+            'last_updated': data['webstore'].get('last_updated'),
+            'name': data['webstore'].get('name'),
+            'offered_by': data['webstore'].get('offered_by'),
+            'permissions_warnings': permission_warnings,
+            'privacy_policy': data['webstore'].get('privacy_policy'),
+            'rating': data['webstore'].get('rating'),
+            'rating_users': data['webstore'].get('rating_users'),
+            'short_description': data['webstore'].get('short_description'),
+            'size': data['webstore'].get('size'),
+            'support_site': data['webstore'].get('support_site'),
+            'users': data['webstore'].get('users'),
+            'website': data['webstore'].get('website'),
+            'type': data['webstore'].get('type'),
+            'price': data['webstore'].get('price')
+        })
+    if len(extensions) == 0:
+        raise ValueError('No extensions could be parsed from JSON data')
+    return extensions
+
+
+def load_extensions(extensions, session, update_tag):
+    """
+    Ingests the extension details into Neo4J
+    :param extensions: List of extension data to load to Neo4J
+    :param session: Neo4J session object for server communication
+    :param update_tag: Timestamp used to determine data freshness
+    :return: None
+    """
+    ingestion_cypher = """
+    UNWIND {ExtensionsData} as extension
+    MERGE (e:ChromeExtension{id: extension.id})
+    ON CREATE SET
+    e.extension_id = extension.extension_id,
+    e.version = extension.version,
+    e.firstseen = timestamp()
+    SET
+    e.extcalls = extension.extcalls,
+    e.risk_total = extension.risk_total,
+    e.risk_metadata = extension.risk_metadata,
+    e.address = extension.address,
+    e.email = extension.email,
+    e.icon = extension.icon,
+    e.last_updated = extension.last_updated,
```

Comment:
[schema-block] let's avoid issues with our internal lastupdated and rename to crxcavator_last_update for example

Reply:
updated